### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by GabrielJSuarez**
Create the whole game logic for the tic tac toe snipppet

### CODE REVIEW

**Review by euqueme — CHANGES_REQUESTED** [review]
> ### Pull request reviewed by Maria Eugenia Queme P. (TSE)

### Hello @GabrielJSuarez and @ricardomonte :lollipop: :balloon: 

### Thank you for your submission :handshake: :handshake: you've done a great job with your game I like that you request for the players' names and that you fill columns and rows instead of just giving a number I think it makes your game stand out from the rest :video_game: :star: that been said let's work on the following changes :construction: :construction_worker: :construction_worker: 

### Behaviour

- [ ] The game is letting the users have empty names
> let's validate that the names aren't empty (and that the user's don't have the same name) or assign a default name when this happens
image

- [ ] Let's add a welcoming message and instructions on how to play the game

> like "This is our Tic tac toe game" instead of straight asking for the player's names, also the instructions could be something like "choose a column from 1 to 3 and a row from 1 to 3"

#### Project Specific

- [ ] No `puts/print` and `get.chomp` outside `main.rb` file. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/only_main_print.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

> all puts/print and get.chomp should be done within the `main.rb` file and not in class methods, so let's refactor your code so this rule is respected

- [ ] Only uses attr_accessors if you really need it and must be aware that by using it they allow writing that property from outside the class. That variable can be set to anything.([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone3/attr_accessor.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

> I changed all your attr_accessors to attr_reader and your game still works fine so please only use them when needed to protect your classes and also save memory.

## [OPTIONAL]

- [ ] a ruby convention is to properly name your files with your classes names

> each of your classes should be declared in a file with their name for example the Display class should be in a file called display.rb let's fix it and do it for your other classes as well

### That's it, team! :raised_hands: 

### Please :pray: do the changes :technologist: :technologist: and submit back for a code review when you are done :memo: 

Happy coding! :rocket: 

Any question? ask me on slack **@ Maru** :woman_technologist: 

PROJECT STATUS: REQUESTED CHANGES :vertical_traffic_light:

File: lib/logic.rb [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      puts "#{name} wins this round!"
+      puts new_game.tabletop
+      true
+    elsif draw == true
+      puts 'Draw!'
+      puts new_game.tabletop
+      true
+    end
+  end
+end
```
Comment (MuriloRoque) [inline lib/logic.rb]: - [ ] Please refactor this method, as it still contains `puts`, that is only allowed in the `main.rb` file.

**Review by MuriloRoque — CHANGES_REQUESTED** [review]
## Changes requested :red_circle: 

### Hello @GabrielJSuarez and @ricardomonte :wave: 

### Thank you for making almost all of the previously requested changes :+1: 

### You just missed one of them, please check my comment in your code below.

### Kindly make the requested changes and submit for another review, please :pray: 

### I will be on Slack 24/7 in case you have any questions about this review :handshake: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Review by MuriloRoque — APPROVED** [review]
# Status: Approved :green_circle: 

### Hello again @GabrielJSuarez and @ricardomonte :wave: 

### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[Twitter Follow](https://twitter.com/MuriloRoquePai1)

**Pull request merged**